File: vetedge/veterinary/report/care_location_occupancy/care_location_occupancy.js
frappe.query_reports["Care Location Occupancy"] = {
	onload(report) {
		window.vetedgeReportVisibility?.apply(report, "Care Location Occupancy");
	},
	filters: [
        { fieldname: "branch", label: __("Branch"), fieldtype: "Link", options: "Branch" },
        { fieldname: "location_type", label: __("Location Type"), fieldtype: "Select", options: ["", "Ward", "Kennel", "Cage", "ICU", "Isolation", "Recovery", "General"] },
        { fieldname: "status", label: __("Status"), fieldtype: "Select", options: ["", "Available", "Occupied", "Cleaning", "Maintenance", "Inactive"] },
        { fieldname: "include_inactive", label: __("Include Inactive"), fieldtype: "Check" },
        { fieldname: "occupied_only", label: __("Occupied Only"), fieldtype: "Check" },
    ],
};

File: vetedge/veterinary/report/care_location_occupancy/care_location_occupancy.py
from vetedge.services.hospitalisation_reports import get_care_location_occupancy_report
from vetedge.services.report_insights import build_report_summary


def execute(filters=None):
	columns, data = get_care_location_occupancy_report(filters)
	return columns, data, None, None, build_report_summary("Care Location Occupancy", data, filters)
